Library/Homebrew/software_spec.rb

```diff
@@ -234,7 +234,8 @@ def initialize(formula, spec)
   end
 
   def compatible_cellar?
-    cellar == :any || cellar == HOMEBREW_CELLAR.to_s
+    cellar == :any || cellar == HOMEBREW_CELLAR.to_s ||
+    (cellar != :any && File.directory?(cellar) && File.realpath(cellar) == HOMEBREW_CELLAR.to_s)
```

Comment:
Agreed. It looks like `Pathname.new('/usr/local/Cellar').realpath().to_s` should work with Ruby 1.8. (I don't think that resolving a single symlink would work in my case, as `/usr/local` is the symlink instead of `/usr/local/Cellar`. I'm not sure exactly how Ruby's symlink resolution works, but `readlink /usr/local/Cellar` returns nothing on my system.)